File: tasks/11_Container_With_Most_Water.py
# https://leetcode.com/problems/container-with-most-water/
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1

        area = 0

        while l < r:
            area = max(area, (r - l) * min(height[r], height[l]))

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1

        return area

File: tasks/11_Container_With_Most_Water.go
package main

func maxArea(height []int) int {
	l, r := 0, len(height)-1

	area := 0

	for l < r {
		area = max11(area, (r-l)*min(height[r], height[l]))

		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}

	return area
}

func min(a, b int) int {
	if a < b {
		return a
	}

	return b
}

func max11(a, b int) int {
	if a > b {
		return a
	}

	return b
}
